crates/router/src/core/recon.rs

```diff
@@ -0,0 +1,190 @@
+        user_email: domain::UserEmail::from_pii_email(user_email.clone())
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to convert recipient's email to UserEmail")?,
+        recipient_email: domain::UserEmail::from_pii_email(
+            state.conf.recipient_emails.recon.clone(),
+        )
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to convert recipient's email to UserEmail")?,
+        settings: state.conf.clone(),
+        subject: format!(
+            "Dashboard Pro Feature Request by {}",
+            user_email.expose().peek()
+        ),
+    };
+
+    state
+        .email_client
+        .compose_and_send_email(
+            Box::new(email_contents),
+            state.conf.proxy.https_url.as_ref(),
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to compose and send email for ProFeatureRequest [Recon]")
+        .async_and_then(|_| async {
+            let auth = user_with_auth_data.1;
+            let updated_merchant_account = storage::MerchantAccountUpdate::ReconUpdate {
+                recon_status: enums::ReconStatus::Requested,
+            };
+
+            let response = db
+                .update_merchant(
+                    key_manager_state,
+                    auth.merchant_account,
+                    updated_merchant_account,
+                    &auth.key_store,
+                )
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable_lazy(|| {
+                    format!("Failed while updating merchant's recon status: {merchant_id:?}")
+                })?;
+
+            Ok(service_api::ApplicationResponse::Json(
+                recon_api::ReconStatusResponse {
+                    recon_status: response.recon_status,
+                },
+            ))
+        })
+        .await
+}
+
+pub async fn generate_recon_token(
+    state: SessionState,
+    user: authentication::UserFromToken,
+) -> RouterResponse<recon_api::ReconTokenResponse> {
+    let token = authentication::AuthToken::new_token(
+        user.user_id.clone(),
+        user.merchant_id.clone(),
+        user.role_id.clone(),
+        &state.conf,
+        user.org_id.clone(),
+        user.profile_id.clone(),
+    )
+    .await
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable_lazy(|| {
+        format!(
+            "Failed to create recon token for params [user_id, org_id, mid, pid] [{}, {:?}, {:?}, {:?}]",
+            user.user_id, user.org_id, user.merchant_id, user.profile_id,
+        )
+    })?;
+
+    Ok(service_api::ApplicationResponse::Json(
+        recon_api::ReconTokenResponse {
+            token: token.into(),
+        },
+    ))
+}
+
+pub async fn recon_merchant_account_update(
```

Comment:
Do the auth related changes here too for merchant, merchant_key_store and user.

Reply:
Done for merchant + key store, cannot include user as it's not user specific API